server/server.go

```diff
@@ -39,25 +62,89 @@ func (s *server) handleGet(w http.ResponseWriter, key string) {
 		log.Println("Error writing response:", err)
 		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
 	}
+	ss := []struct {
+		name *http.Server
+	}{
+		{name: &http.Server{ReadHeaderTimeout: ti}},
+	}
+	_ = ss
 }
 
-// according to rfc guidlines PUT should create or replace resources
+// according to rfc guidelines PUT should create or replace resources
 // https://www.rfc-editor.org/rfc/rfc2616#section-9.6
 func (s *server) handlePut(w http.ResponseWriter, r *http.Request, key string) {
-	_, ok := s.db.get(key)
-	if !ok {
-		w.WriteHeader(http.StatusCreated)
-	} else {
-		w.WriteHeader(http.StatusOK)
-	}
 	body, err := io.ReadAll(r.Body)
 	if err != nil {
 		log.Println("Error reading body:", err)
 		http.Error(w, "Error reading body", http.StatusBadRequest)
+		return
+	}
+	code, err := s.db.put(key, string(body))
+	var keyErr *KeyError
+	var valueErr *ValueError
+	var dbErr *DatabaseError
+	switch {
+	case errors.As(err, &keyErr):
+		w.WriteHeader(http.StatusRequestEntityTooLarge)
+		_, err = w.Write([]byte(err.Error()))
+		if err != nil {
+			log.Println(err)
+		}
+		return
+	case errors.As(err, &valueErr):
+		w.WriteHeader(http.StatusRequestEntityTooLarge)
+		_, err = w.Write([]byte(err.Error()))
+		if err != nil {
+			log.Println(err)
```

Comment:
Using slog for client, but not for server?

Reply:
@avorima old habits die hard. Started with log but then realized slog is the way to go. Refactoring now.